cmd/drone-server/server.go

```diff
@@ -675,6 +675,18 @@ func (a *authorizer) authorize(ctx context.Context) error {
 	return errors.New("missing agent token")
 }
 
+func redirect(w http.ResponseWriter, req *http.Request) {
+	var serverHost string = droneserver.Config.Server.Host
+	serverHost = strings.TrimPrefix(serverHost, "http://")
+	serverHost = strings.TrimPrefix(serverHost, "https://")
+	req.URL.Scheme = "https"
+	req.URL.Host = serverHost
+
+	w.Header().Set("Strict-Transport-Security", "max-age=31536000")
```

Comment:
We already had HSTS header in `Secure` middleware : https://github.com/drone/drone/blob/master/router/middleware/header/header.go#L38

Reply:
The secure middleware is currently being applied to gin routes and therefore the header would not be added to a `net/http` handler: https://github.com/drone/drone/blob/bb8c569249db8fe42cac7fee6d8c50e3b46b68ca/router/router.go#L26

In addition, the current middleware only sends the header to requests that already have TLS enabled:
https://github.com/drone/drone/blob/8eacbe382b73f4f51f2d2042086a469f2e6a8028/router/middleware/header/header.go#L43-L45

Reply:
I will improve this a bit later when I'm refactoring some of our routing code in 0.9

Reply:
I mean `Strict-Transport-Security` header is no need for http response, just need for https response (after redirecting)

Reply:
That's a good point @khanhicetea. I actually didn't know the RFC mandates that the header is ignored on non-secure requests, which means setting the header is actually useless:
> If an HTTP response is received over insecure transport, the UA
      MUST ignore any present STS header field(s).

https://tools.ietf.org/html/rfc6797#section-8.1